File: skellig-test-step-runner/src/main/java/org/skellig/teststep/runner/exception/TestStepRegistryException.java
package org.skellig.teststep.runner.exception;

public class TestStepRegistryException extends RuntimeException {

    public TestStepRegistryException(String message) {
        super(message);
    }

    public TestStepRegistryException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: skellig-test-step-runner/src/main/kotlin/org/skellig/teststep/runner/exception/TestStepRegistryException.kt
package org.skellig.teststep.runner.exception

/**
 * Exception thrown by the TestStepRegistry class when registration of a test step fails.
 *
 * @param message The detail message.
 * @param cause The cause of the exception.
 */
class TestStepRegistryException(message: String?, cause: Throwable?) : RuntimeException(message, cause) {
    constructor(message: String?) : this(message, null)
}